server/service/src/stocktake/insert/generate.rs

```diff
@@ -201,3 +217,110 @@ fn generate_lines_initial_stocktake(
 
     Ok(result)
 }
+
+pub fn generate_lines_from_master_list(
+    connection: &StorageConnection,
+    store_id: &str,
+    stocktake_id: &str,
+    master_list_id: &str,
+) -> Result<Vec<StocktakeLineRow>, RepositoryError> {
+    let item_ids: Vec<String> = MasterListLineRepository::new(connection)
+        .query_by_filter(
+            MasterListLineFilter::new()
+                .master_list_id(EqualFilter::equal_to(master_list_id))
+                .item_type(ItemType::Stock.equal_to()),
+        )?
+        .into_iter()
+        .map(|r| r.item_id)
+        .collect();
+
+    let mut result = Vec::<StocktakeLineRow>::new();
+
+    item_ids.iter().for_each(|item_id| {
+        let stock_lines = StockLineRepository::new(connection)
+            .query_by_filter(
+                StockLineFilter::new()
+                    .item_id(EqualFilter::equal_to(item_id))
+                    .store_id(EqualFilter::equal_to(store_id))
+                    .has_packs_in_store(true),
+                Some(store_id.to_string()),
+            )
+            .unwrap();
+        let item_name = ItemRowRepository::new(connection)
+            .find_active_by_id(item_id)
+            .unwrap()
```

Comment:
Don't like these unwraps, should I handle them better?

Reply:
.find_active_by_id(item_id)?
            .map(|item_row| item_row.name);